lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -954,7 +954,10 @@ def headers(options)
         cookie = options[:cookie] || cookie
         headers['Cookie'] = cookie if cookie
 
-        headers['Idempotency-Key'] = idempotency_key if idempotency_key
+        if idempotency_key
+          headers['Idempotency-Key'] = idempotency_key
+          options[:idempotency_key] = SecureRandom.uuid
```

Comment:
Could you just add a note to the PR to prevent confusion on what this does?

Reply:
added! Thanks!